vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java

```diff
@@ -231,6 +237,106 @@ public void findBlobs(StringComparisonRule rule, String blobNameToMatch, String
+     *                          </code>
+     *                          <br>
+     *                          <br>where all filters are optional, but at least one rule is required.
+     *                          <ul>
+     *                          <li><code>blobNameFilterRule</code> The blob name comparison rule: "is equal to",
+     *                           "contains", "does not contain" or "matches".
+     *                           Should be specified along with <i>blobNameFilterValue</i>.</li>
+     *                          <li><code>blobNameFilterValue</code> The full or partial blob name to be matched.
+     *                           Should be specified along with <i>blobNameFilterRule</i>.</li>
+     *                          <li><code>blobNamePrefix</code> The prefix which blob names should start with.</li>
+     *                          <li><code>resultsLimit</code> Maximum number of blob names to return.</li>
+     *                          </ul>
+     * @param containerName     The name of the container to point to.
+     * @param storageAccountKey The key to Storage Account endpoint.
+     * @param scopes            The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of the variable
+     *                          scopes.<br>
+     *                          <i>Available scopes:</i>
+     *                          <ul>
+     *                          <li><b>STEP</b> - the variable will be available only within the step,
+     *                          <li><b>SCENARIO</b> - the variable will be available only within the scenario,
+     *                          <li><b>STORY</b> - the variable will be available within the whole story,
+     *                          <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     *                          </ul>
+     * @param variableName      The variable name to store the list of found blob names.
+     */
+    @When("I filter blobs by:$filter in container `$containerName` of storage account `$storageAccountKey`"
+            + " and save result to $scopes variable `$variableName`")
+    public void findBlobsByFilter(BlobFilter filter, String containerName,
+                                      String storageAccountKey, Set<VariableScope> scopes, String variableName)
+    {
+        filter.validate();
+        BlobContainerClient blobContainerClient = createBlobContainerClient(containerName, storageAccountKey);
+        bddVariableContext.putVariable(scopes, variableName, getLimitedBlobNames(blobContainerClient, filter));
+    }
+
+    private static List<String> getLimitedBlobNames(BlobContainerClient blobContainerClient, BlobFilter filter)
+    {
+        return filter.getResultsLimit()
+                .map(limit -> {
+                    ListBlobsOptions options = new ListBlobsOptions();
+                    options.setMaxResultsPerPage(limit.intValue());
```

Comment:
Is there a server limitation? I think it is expected that we could receive only a limited maximum results per page